### DISCUSSION TITLE
How should I use Orchard Core for complex interactions

### DISCUSSION THREAD

**Discussion by ChestnutYYY** [OrchardCMS/OrchardCore#11342] [Q&A]
Orchard Core Decoupled CMS

I made a page with a button, and when i click the button,  I would like to be able to have a popup or be able to have an ajax request or be able to control whether other modules are shown or not.  May be , These functions can be implemented with jQuery.  But not very friendly.  Is there any other way please?  I need  help.

**Comment by hishamco** [comment]
You can use AJAX request through jQuery or JavaScript, then let an MVC Controller or RazorPage Model that handle that request

**Comment by ns8482e** [comment]
For decoupled, you can use Orchard helper in MVC or Razor pages to interact with orchard data 

For SPA/client side with angular/react/vue/jquery you can use openid and orchard content api or graphql to interact with orchard data

**Reply by ChestnutYYY** [reply]
As far as I know, there is a built-in content part that is button, and I want to combine button events with react components or vue components. I wonder if this is possible?

**Reply by ns8482e** [reply]
`Button` content item is from the  forms module, the module allows you to create dynamic forms and uses orchard cms’s display(so it won’t be decoupled)

You can create a controller/page and provide that in action of the form to handle form submission, and/or you can use client side script to post that form via Ajax

you can also use workflow

**Reply by ChestnutYYY** [reply]
okey, Thks.
